src/libs/OptionsListUtils.js

```diff
@@ -618,15 +668,71 @@ function getOptions(
         includeMoneyRequests = false,
         excludeUnknownUsers = false,
         includeP2P = true,
+        includeCategories = false,
+        categories = {},
+        recentlyUsedCategories = {},
         canInviteUser = true,
     },
 ) {
+    if (includeCategories) {
+        const categoryOptions = [];
+
+        const categoriesAmount = _.size(categories);
+
+        if (!_.isEmpty(searchInputValue)) {
+            categoryOptions.push({
+                title: '', // Search result
+                data: getOptionTree(
+                    _.filter(categories, (category) => category.name.toLowerCase().includes(searchInputValue.toLowerCase())),
+                    true,
+                ),
+            });
+        } else if (categoriesAmount < CONST.CATEGORY_LIST_THRESHOLD) {
+            categoryOptions.push({
+                title: '', // All
+                data: getOptionTree(categories),
+            });
+        } else {
+            const selectedOptionNames = _.map(selectedOptions, (selectedOption) => selectedOption.name);
+            const filteredRecentlyUsedCategories = _.filter(recentlyUsedCategories, (category) => !_.includes(selectedOptionNames, category.name));
+            const filteredCategories = _.filter(categories, (category) => !_.includes(selectedOptionNames, category.name));
+
+            if (!_.isEmpty(selectedOptions)) {
+                categoryOptions.push({
+                    title: '', // Selected
+                    data: getOptionTree(selectedOptions),
+                });
+            }
+
+            if (!_.isEmpty(filteredRecentlyUsedCategories)) {
+                categoryOptions.push({
+                    title: 'Recent',
+                    data: getOptionTree(filteredRecentlyUsedCategories.slice(0, maxRecentReportsToShow)),
+                });
+            }
+
+            categoryOptions.push({
+                title: 'All',
+                data: getOptionTree(filteredCategories),
+            });
+        }
+
+        return {
+            recentReports: [],
+            personalDetails: [],
+            userToInvite: null,
+            currentUserOption: null,
+            categoryOptions,
+        };
+    }
```

Comment:
This is just a suggestion, but perhaps I would extract this to a function. `getOptions` is huge, and since this is an early return, perhaps it would be better to move this logic to a function with a descriptive name.

Reply:
Good point! I will ask someone from Expensify, because they planned initially to have it here. However, I agree with you.

Reply:
@sobitneupane @puneetlath wdyt?

Reply:
I agree, that seems better. 

@yuwenmemon I'm guessing you wrote the plan in the doc. What do you think?

Reply:
Done - 657850f.